cpp/src/parquet/metadata.cc

```diff
@@ -87,8 +87,8 @@ std::string ParquetVersionToString(ParquetVersion::type ver) {
 template <typename DType>
 static std::shared_ptr<Statistics> MakeTypedColumnStats(
     const format::ColumnMetaData& metadata, const ColumnDescriptor* descr) {
-  // If ColumnOrder is defined, return max_value and min_value
-  if (descr->column_order().get_order() == ColumnOrder::TYPE_DEFINED_ORDER) {
+  // Return max_value and min_value if defined
+  if (metadata.statistics.__isset.max_value || metadata.statistics.__isset.min_value) {
```

Comment:
I don't think `ColumnOrder` checking can be removed here. If a writer does not set ColumnOrder but writes both min/max and min_value/max_value, we should still get stats from min/max as per the specs.